hoot-core/src/main/cpp/hoot/core/algorithms/linearreference/WaySublineRemover.cpp

```diff
@@ -60,10 +63,28 @@ std::vector<ElementId> WaySublineRemover::removeSubline(
   if (subline.getStart().isExtreme() && subline.getEnd().isExtreme())
   {
     LOG_TRACE(
-      "Subline matches covers entire way; removing entire way: " << way->getElementId());
-    // Use RemoveElementByEid here instead of RecursiveElementRemover so that the way is removed
-    // from its parent before its removal.
-    RemoveElementByEid(way->getElementId()).apply(map);
+      "Subline matches covers entire way. Removing entire way: " << way->getElementId() << "...");
+
+    if (removeWholeWayRecursively)
+    {
+      // remove it from parents and the map
+      RemoveElementByEid(way->getElementId()).apply(map);
+      // remove its children
+      RecursiveElementRemover(way->getElementId()).apply(map);
+    }
+    else
+    {
+      // Use RemoveElementByEid here instead of RecursiveElementRemover so that the way is removed
+      // from its parent before its removal. We also don't want to remove RecursiveElementRemover here
+      // b/c the way's nodes may still belong to one the temp relations we created during splitting.
+      RemoveElementByEid(way->getElementId()).apply(map);
+    }
+
+    if (ConfigOptions().getDebugMapsWrite() && ConfigOptions().getDebugMapsWriteDetailed())
+    {
```

Comment:
Added a new option, `debug.maps.write.detailed` to go with `debug.maps.write`. The detailed method is meant for debug maps written in a loop, where a ton of them could get output for a conflate job.